docs/competitive-programming/prefix-sum-algorithm.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: Prefix Sum Algorithm
+slug: /competitive-programming/prefix-sum-algorithm
+---
+
+## Introduction
+
+You are given an array of `N` elements. It's Prefix Sum is another array of same size.
+where PrefixSum[i] is sum of all prefix elements of the given array. Formally, 
+PrefixSum[i] = ar[0] `+` ar[1] `+` ar[2]... ar[i].
+
+
+## Explanation
+
+Let us have one example:
+
+```
+Let us say, given array is ar[] = {1,2,3,4};
+then it's prefix sum array will be prefix_sum_array[] = {1,3,6,10};
+here, prefix_sum_array[1] = ar[0] `+` ar[1].
+thus will goes on for the rest indexes of Prefix Sum array.
+```
+
+## Algorithm
+
+ - New Prefix Sum array's first element will be same as given array's first element.
+ - Next we will run a loop from index `1` to `last` index.
+ - In Each iteration, Prefix Sum array's current element will be sum of previous element of this array and current index value of our given array.
+ - So, iterations will be like this, prefix_sum_array[1] = prefix_sum_array[0] `+` ar[1].
+ - prefix_sum_array[2] = prefix_sum_array[1] `+` ar[2] and so on.
```

Comment:
Please put `prefix_sum_array[2]` and other similar code instances in backticks as well.

Reply:
@vybhav72954  you mean to say add prefix_sum_array[3] , [4] , [5] till n?
each iteration?

Reply:
`prefix_sum_array[2]` = `prefix_sum_array[1] + ar[2]` and so on.

So, iterations will be like this, `prefix_sum_array[1]` = `prefix_sum_array[0] + ar[1]`.